src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -797,7 +797,7 @@ impl BatchPlanFragmenter {
                     )
                 } else {
                     // can be 0 if no available serving worker
-                    self.worker_node_manager.worker_node_count()
+                    self.worker_node_manager.schedule_unit_count()
```

Comment:
This is equal to summary of number of cpu cores on worker nodes? This may be too large, maybe `self.worker_node_manager.schedule_unit_count() /2 ` would be enough?